### PR TITLE
feat(anonymizer): allow expression language as dynamic value

### PR DESCRIPTION
**Pull request by maxperei**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?   | no
| Fixed issue | #33

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -95,7 +95,20 @@ Sylius\Component\Core\Model\Address: # Your class path
          prefix: 'anonymized-'
  ```
 
-   Value can be null, an array, an int and a string
+   `value` can be null, an array, an int, a string and an expression language
```
Comment (oallain) [inline README.md]:
```suggestion
   `value` can be null, an array, an int, a string and an [expression language](https://symfony.com/doc/current/reference/formats/expression_language.html) 
```

File: src/Provider/Anonymizer.php [hunk 2]
```diff
@@ -102,6 +106,9 @@ public function anonymize(Object $entity, bool $reset = false, int $maxRetries =
         $this->eventDispatcher->dispatch(new AfterAnonymize($entity, ['entity' => $clonedEntity]));
     }
 
+    /**
+     * @SuppressWarnings(PHPMD.NPathComplexity)
```
Comment (oallain) [inline src/Provider/Anonymizer.php]: I don't know if it's easy to do, but can you try extracting parts of the method to remove this?

File: tests/PHPUnit/config/mappings/anon1.yml [hunk 3]
```diff
@@ -7,4 +7,6 @@ Tests\Synolia\SyliusGDPRPlugin\PHPUnit\Fixtures\YamlFoo:
         value:
             value: 'test-yaml-value'
         nullValue:
-            value: null
\ No newline at end of file
+            value: null
+        dynamicValue:
+            value: '@="anonymized-" ~ object.getId() ~ "@domain+1.xyz"'
```
Comment (oallain) [inline tests/PHPUnit/config/mappings/anon1.yml]: Can you add blank line at the end of this file ?

File: tests/PHPUnit/config/mappings/anon2.yml [hunk 4]
```diff
@@ -10,4 +10,4 @@ Tests\Synolia\SyliusGDPRPlugin\PHPUnit\Fixtures\YamlFoo:
             prefix: 'test-yaml-prefix-'
         prefixValue:
             value: value
-            prefix: 'test-yaml-prefix-'
+            prefix: 'test-yaml-prefix-'
```
Comment (oallain) [inline tests/PHPUnit/config/mappings/anon2.yml]: Can you add blank line at the end of this file ?

File: src/Provider/Anonymizer.php (hunk 2 above)
Comment (maxperei) [reply src/Provider/Anonymizer.php:110]: no it's not, to do so we have to rewrite this method entirely

**Review by oallain — APPROVED** [review]
Looks good 💪🏼

**Pull request merged**